galasa-ui/src/utils/3270/get3270Screenshots.ts

```diff
@@ -107,32 +108,39 @@ export function populateFlattenedZos3270TerminalDataAndAllImageData(images: Term
 }
 
 export const get3270Screenshots = async (zos3270TerminalData: TreeNodeData[], runId: string) => {
-  for (var terminal of zos3270TerminalData) {
-    const zippedFilesContainingImageJSON: FileNode[] = Object.values(terminal.children)
-      .filter((node) => (node as FileNode).isFile)
-      .map((node) => node as FileNode);
-
-    for (var file of zippedFilesContainingImageJSON) {
-      await downloadArtifactFromServer(runId, file.url).then((artifactData) => {
-        // Unzip the content
-        const images: TerminalImage[] = unzipBase64(artifactData).images;
-
-        populateFlattenedZos3270TerminalDataAndAllImageData(images);
-      });
+  // Test if user is requesting same resource as the previously fetched one, if so then simply return existing value.
+  if (
+    newAllImageData.length === 0 ||
+    newFlattenedZos3270TerminalData.length === 0 ||
+    previousRunId !== runId
+  ) {
+    newAllImageData.length = 0;
+    newFlattenedZos3270TerminalData.length = 0;
+
+    for (var terminal of zos3270TerminalData) {
+      const zippedFilesContainingImageJSON: FileNode[] = Object.values(terminal.children)
+        .filter((node) => (node as FileNode).isFile)
+        .map((node) => node as FileNode);
```

Comment:
Minor - could the `map` and `filter` operations be swapped around so that we don't need the `as FileNode` bit twice?

Something like this:

```suggestion
        .map((node) => node as FileNode)
        .filter((node) => node.isFile);
```

Reply:
Great idea - done now thank you